core/src/test/scala/org/broadinstitute/dsde/rawls/monitor/MultiregionalBucketMigrationActorSpec.scala

```diff
@@ -0,0 +1,915 @@
+              GcsBucketName("tmp-bucket-name").some
+            )
+          } yield ()
+        }
+
+        mockSts = new MockStorageTransferService {
+          override def createTransferJob(jobName: JobName,
+                                         jobDescription: String,
+                                         projectToBill: GoogleProject,
+                                         sourceBucket: GcsBucketName,
+                                         destinationBucket: GcsBucketName,
+                                         schedule: JobTransferSchedule,
+                                         options: Option[GoogleStorageTransferService.JobTransferOptions]
+          ) =
+            getStsServiceAccount(projectToBill).flatMap { serviceAccount =>
+              IO.raiseError(
+                new StatusRuntimeException(
+                  Status.FAILED_PRECONDITION.withDescription(
+                    s"Service account ${serviceAccount.email} does not have required " +
+                      "permissions {storage.objects.create, storage.objects.list} " +
+                      s"for bucket $destinationBucket."
+                  )
+                )
+              )
+            }
+        }
+
+        _ <- MigrateAction.local(_.copy(storageTransferService = mockSts))(migrate)
+
+        before <- inTransaction { dataAccess =>
+          @nowarn("msg=not.*?exhaustive")
+          val update = for {
+            Some(migration) <- dataAccess.multiregionalBucketMigrationQuery
+              .getAttempt(testData.workspace.workspaceIdAsUUID)
+              .value
+            retry <- dataAccess.multiregionalBucketMigrationRetryQuery.getOrCreate(migration.id)
+            _ <- dataAccess.multiregionalBucketMigrationRetryQuery.update(
+              retry.id,
+              dataAccess.multiregionalBucketMigrationRetryQuery.retriesCol,
+              1L
+            )
+          } yield migration
+          update
+        }
+
+        _ <- migrate
+        _ <- inTransaction { dataAccess =>
+          @nowarn("msg=not.*?exhaustive")
+          val test = for {
+            Some(after) <- dataAccess.multiregionalBucketMigrationQuery.getAttempt(before.id).value
+            transferJobs <- storageTransferJobs.filter(_.migrationId === before.id).result
+          } yield {
+            transferJobs shouldBe empty
+            after shouldBe before
+          }
+          test
+        }
+      } yield succeed
+    }
+
+  it should "not prevent a workspace from being deleted if the migration was retried" in
+    runMigrationTest {
+      for {
+        _ <- inTransaction { dataAccess =>
+          createAndScheduleWorkspace(testData.workspace) >>=
+            dataAccess.multiregionalBucketMigrationRetryQuery.getOrCreate
+        }
+        wsService <- MigrateAction.asks(_.workspaceService)
+        _ <- MigrateAction.fromFuture {
+          wsService.deleteWorkspace(testData.workspace.toWorkspaceName)
```

Comment:
What is the failure mode here? The workspace is locked and therefore undeletable?

Reply:
[Relevant ticket](https://broadworkbench.atlassian.net/browse/PPWM-105)

There was a problem in PPW where cascade delete wasn't set up on the FK from the retry table to the migration history table, so workspaces that had retried migrations couldn't be deleted as the FK constraint prevented it